tensorflow/lite/micro/kernels/ceva_bx1/fully_connected.cc

```diff
@@ -0,0 +1,348 @@
+  int8_t* inputp = (int8_t*)tflite::micro::GetTensorData<int8_t>(input);
+  int8_t* filterp = (int8_t*)tflite::micro::GetTensorData<int8_t>(filter);
+  int32_t* biasp = (int32_t*)tflite::micro::GetTensorData<int32_t>(bias);
+  int8_t* outputp = (int8_t*)tflite::micro::GetTensorData<int8_t>(output);
+#ifdef MCPS_MEASUREMENT
+  int batches = output_shape_DimsData[0];
+  int output_depth = weights_shape_DimsData[weights_shape_DimensionsCount - 2];
+  int accum_depth = weights_shape_DimsData[weights_shape_DimensionsCount - 1];
+  MCPS_START_ONE;
+#endif
+  CEVA_TFLM_FullyConnected_int8(
+      params,
+      input_shape_DimensionsCount,  // GetTensorShape(input),
+      inputp,
+      weights_shape_DimensionsCount,  // GetTensorShape(filter),
+      weights_shape_DimsData, filterp,
+      bias_shape_DimensionsCount,  // GetTensorShape(bias),
+      biasp,
+      output_shape_DimensionsCount,  // GetTensorShape(output),
+      output_shape_DimsData, outputp);
+#ifdef MCPS_MEASUREMENT
+  MCPS_STOP_ONE(
+      "Test params:Call CEVA_TFLM_FullyConnected_int8 inetrnal loop = %dx%dx%d",
+      batches, output_depth, accum_depth);
+#endif
+
+#endif  // ORIGINAL_IMPLEMENTATION
+
+  return kTfLiteOk;
+}
+
+TfLiteStatus EvalQuantized(TfLiteContext* context, TfLiteNode* node,
+                           const OpData& data, const TfLiteEvalTensor* input,
+                           const TfLiteEvalTensor* filter,
+                           const TfLiteEvalTensor* bias,
+                           TfLiteEvalTensor* output) {
+  const int32_t input_offset = -data.input_zero_point;
+  const int32_t filter_offset = -data.filter_zero_point;
+  const int32_t output_offset = data.output_zero_point;
+
+  tflite::FullyConnectedParams op_params;
+  op_params.input_offset = input_offset;
+  op_params.weights_offset = filter_offset;
+  op_params.output_offset = output_offset;
+  op_params.output_multiplier = data.output_multiplier;
+  // Legacy ops used mixed left and right shifts. Now all are +ve-means-left.
+  op_params.output_shift = -data.output_shift;
+  op_params.quantized_activation_min = data.output_activation_min;
+  op_params.quantized_activation_max = data.output_activation_max;
+
+#define TF_LITE_FULLY_CONNECTED(output_data_type)      \
+  reference_ops::FullyConnected(                       \
+      op_params, tflite::micro::GetTensorShape(input), \
+      tflite::micro::GetTensorData<uint8_t>(input),    \
+      tflite::micro::GetTensorShape(filter),           \
+      tflite::micro::GetTensorData<uint8_t>(filter),   \
+      tflite::micro::GetTensorShape(bias),             \
+      tflite::micro::GetTensorData<int32_t>(bias),     \
+      tflite::micro::GetTensorShape(output),           \
+      tflite::micro::GetTensorData<output_data_type>(output))
+  switch (output->type) {
+    case kTfLiteUInt8:
+      TF_LITE_FULLY_CONNECTED(uint8_t);
+      break;
+    case kTfLiteInt16:
+      TF_LITE_FULLY_CONNECTED(int16_t);
+      break;

```

Comment:
I would suggest removing int16 support as well. Unless you have a model that needs it. We're not decided how far we want to go with int16 for TFLM at this time.